plugin.video.hrmagplayer/libplayer/player.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import sys
+from libplayer.channel import *
+from libplayer.utils import *
+
+# Get the plugin url in plugin:// notation.
+__url__ = sys.argv[0]
+# Get the plugin handle as an integer number.
+__handle__ = int(sys.argv[1])
+
+__context__ = None
+
+class HrMagPlayer(xbmc.Player):
+    def __init__(self):
+        print 'HrMagPlayer initialized'
+        xbmc.Player.__init__(self)
+    
+    def onPlayBackError(self):
+        logStr = 'onPlayBackError called for Hr3Player'
+        print logStr
+        xbmc.log(logStr, xbmc.LOGERROR)
+    
+def list_shows(context):
+    """
+    Create the list of video categories in the Kodi interface.
+    :return: None
+    """
+    # Get video categories
+    shows = context['showList']
+    # Create a list for our items.
+    listing = []
+    # Iterate through categories
+    index = 0
+    for show in shows:
+        if show['active']:
+            # Create a list item with a text label and a thumbnail image.
+            list_item = xbmcgui.ListItem(label=show['name'], thumbnailImage=show['image'])
```

Comment:
`thumbnailImage` kwarg is deprecated. Please use `list_item.setArt({"thumb": "xxx", "icon": "xxx", "fanart":"xxx" ...etc })` instead